### PR TITLE
fabtests/efa: add fork-related test

### PR DESCRIPTION
**Pull request by shijin-aws**
Run fi_rdm_tagged_bw with "-K" with environment variables
` FI_EFA_FORK_SAFE=1` and `RDMAV_FORK_SAFE=1`

### CODE REVIEW

File: fabtests/pytest/efa/test_rdm.py [hunk 1]
```diff
@@ -56,3 +56,21 @@ def test_rdm_atomic(cmdline_args, iteration_type, completion_type):
                             timeout=1800)
     test.run()
 
+
+@pytest.mark.functional
+@pytest.mark.parametrize("environment_variable", ["FI_EFA_FORK_SAFE", "RDMAV_FORK_SAFE"])
+def test_rdm_fork(cmdline_args, completion_type, environment_variable):
```
Comment (wzamazon) [inline fabtests/pytest/efa/test_rdm.py:62]: I do not think it belong's to `test_rdm.py`, fork support show be for any endpoint type.  maybe rename `test_fork_support` and put it in `test_fork_support.py`

File: fabtests/pytest/efa/test_rdm.py [hunk 2]
```diff
@@ -56,3 +56,21 @@ def test_rdm_atomic(cmdline_args, iteration_type, completion_type):
...
+    if cmdline_args_copy.environments:
+        cmdline_args_copy.environments += " "
+    else:
+        cmdline_args_copy.environments = ""
+
+    cmdline_args_copy.environments += "{}=1".format(environment_variable)
+    test = ClientServerTest(cmdline_args_copy, "fi_rdm_tagged_bw -K",
+                            completion_type=completion_type,
+                            datacheck_type="with_datacheck")
+    test.run()
```
Comment (wzamazon) [inline fabtests/pytest/efa/test_rdm.py:76]: Added a new line at end?

**Review by wzamazon — COMMENTED** [review]
Would it be possible to added negative test, that when the environment variable is not set efa will abort and print warning?

I understand that would require calling `ibv_is_fork_initialized()` and check the return against `IBV_FORK_UNNEEDED`.

**Comment by shijin-aws** [comment]
> 



> Would it be possible to added negative test, that when the environment variable is not set efa will abort and print warning?
> 
> I understand that would require calling `ibv_is_fork_initialized()` and check the return against `IBV_FORK_UNNEEDED`.

I thought about that, and as you said, for the negative test we need to check kernel's fork support via rdma-core API, which is not straightforward to use in fabtests. We may add that test in the future after we figure out how to use pyverbs

**Review by wzamazon — APPROVED** [review]

**Review by wzamazon — COMMENTED** [review]
Thanks! The other 2 comments still apply.

**Review by wzamazon — CHANGES_REQUESTED** [review]
the other two comments still apply

File: fabtests/pytest/efa/test_rdm.py (hunk 1 above)
Comment (shijin-aws) [reply fabtests/pytest/efa/test_rdm.py]: I added a new file named `test_fork.py` to have this test.

File: fabtests/pytest/efa/test_rdm.py (hunk 2 above)
Comment (shijin-aws) [reply fabtests/pytest/efa/test_rdm.py]: Fixed

File: fabtests/pytest/efa/test_fork.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import pytest
+
+
+@pytest.mark.functional
+@pytest.mark.parametrize("environment_variable", ["FI_EFA_FORK_SAFE", "RDMAV_FORK_SAFE"])
+def test_rdm_fork(cmdline_args, completion_type, environment_variable):
```
Comment (wzamazon) [inline fabtests/pytest/efa/test_fork.py:6]: I think the function name and file name should be `test_fork_support`. 

test_fork is too ambiguous.

Comment (shijin-aws) [reply fabtests/pytest/efa/test_fork.py]: Done

**Review by wzamazon — APPROVED** [review]

**Pull request merged**